pyro/compressible/simulation.py

```diff
@@ -193,33 +207,105 @@ def evolve(self):
+                                                           self.cc_data, self.rp, self.ivars,
+                                                           self.solid, self.tc, self.dt)
+
+        # Get the actual interface conserved state after using Riemann Solver
+        # Then construct the corresponding fluxes using the conserved states
+
+        if myg.coord_type == 1:
+            # We need pressure from interface state for conservative update for
+            # SphericalPolar geometry. So we need interface conserved states.
+
+            F_x, U_x = riemann.riemann_flux(1, U_xl, U_xr,
+                                            self.cc_data, self.rp, self.ivars,
+                                            self.solid.xl, self.solid.xr, self.tc,
+                                            return_cons=True)
+
+            F_y, U_y = riemann.riemann_flux(2, U_yl, U_yr,
+                                            self.cc_data, self.rp, self.ivars,
+                                            self.solid.yl, self.solid.yr, self.tc,
+                                            return_cons=True)
+
+            # Find primitive variable since we need pressure in conservative update.
+            qx = cons_to_prim(U_x, gamma, self.ivars, myg)
+            qy = cons_to_prim(U_y, gamma, self.ivars, myg)
+
+        else:
+            # Directly calculate the interface flux using Riemann Solver
+            F_x = riemann.riemann_flux(1, U_xl, U_xr,
+                                       self.cc_data, self.rp, self.ivars,
+                                       self.solid.xl, self.solid.xr, self.tc,
+                                       return_cons=False)
+
+            F_y = riemann.riemann_flux(2, U_yl, U_yr,
+                                       self.cc_data, self.rp, self.ivars,
+                                       self.solid.yl, self.solid.yr, self.tc,
+                                       return_cons=False)
+
+        # Apply artificial viscosity to fluxes
+
+        q = cons_to_prim(self.cc_data.data, gamma, self.ivars, myg)
+
+        F_x, F_y = flx.apply_artificial_viscosity(F_x, F_y, q,
+                                                  self.cc_data, self.rp,
+                                                  self.ivars)
 
         old_dens = dens.copy()
+        old_xmom = xmom.copy()
         old_ymom = ymom.copy()
 
         # conservative update
-        dtdx = self.dt/myg.dx
-        dtdy = self.dt/myg.dy
+        dtdV = self.dt / myg.V.v()
 
         for n in range(self.ivars.nvar):
             var = self.cc_data.get_var_by_index(n)
 
-            var.v()[:, :] += \
-                dtdx*(Flux_x.v(n=n) - Flux_x.ip(1, n=n)) + \
-                dtdy*(Flux_y.v(n=n) - Flux_y.jp(1, n=n))
+            var.v()[:, :] += dtdV * \
+                (F_x.v(n=n)*myg.Ax.v() - F_x.ip(1, n=n)*myg.Ax.ip(1) +
+                 F_y.v(n=n)*myg.Ay.v() - F_y.jp(1, n=n)*myg.Ay.jp(1))
+
```

Comment:
I don't think that this is right.  Only the divergence terms should have the area factors.  For non-Cartesian, the grad{p} term is different and we need to address that separately.

Reply:
gradp is added after this , where I apply external source. For non cartesian, xmom and ymom flux don't have pressure, you can check consFlux.

Reply:
ah okay.  Can you update the comment then to say that it also adds grad p for the coord_type = 1 case?

Reply:
added more comments.